influxdb3_cache/src/last_cache/cache.rs

```diff
@@ -340,32 +339,36 @@ impl LastCache {
     /// This will panic if the internal cache state's keys are out-of-order with respect to the
     /// order of the `key_columns` on this [`LastCache`]
     pub(crate) fn push(&mut self, row: &Row, table_def: Arc<TableDefinition>) {
-        let accept_new_fields = self.accept_new_fields();
-        let mut target = &mut self.state;
-        let mut key_iter = self.key_column_ids.iter().peekable();
-        while let (Some(col_id), peek) = (key_iter.next(), key_iter.peek()) {
-            if target.is_init() {
-                *target = LastCacheState::Key(LastCacheKey {
-                    column_id: *col_id,
-                    value_map: Default::default(),
-                });
-            }
+        let mut values = Vec::with_capacity(self.key_column_ids.len());
+        for id in self.key_column_ids.iter() {
             let Some(value) = row
                 .fields
                 .iter()
-                .find(|f| f.id == *col_id)
+                .find(|f| &f.id == id)
                 .map(|f| KeyValue::from(&f.value))
             else {
                 // ignore the row if it does not contain all key columns
                 return;
             };
+            values.push(value);
+        }
```

Comment:
This makes it so that the incoming `Row` is validated for all key columns before updating the state of the cache, so that an incomplete row won't result in an incomplete state update.